File: latest_ran_output/qwen3_6_27b/file_004_naive.py
import os
import subprocess


def resize_image(input_path: str, output_path: str, width: int = 800, height: int = 600) -> bool:
    """
    Resizes an image using the ImageMagick 'convert' shell command.
    
    Args:
        input_path (str): Path to the source image file.
        output_path (str): Path where the resized image will be saved.
        width (int): Target width in pixels.
        height (int): Target height in pixels.
        
    Returns:
        bool: True if successful, False otherwise.
    """
    if not os.path.exists(input_path):
        return False

    try:
        # Use ImageMagick's convert command to resize the image
        # -resize WxH maintains aspect ratio by default unless ! is added
        cmd = [
            "convert",
            input_path,
            "-resize", f"{width}x{height}",
            output_path
        ]
        
        subprocess.run(cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        return True
    except FileNotFoundError:
        # ImageMagick might not be installed or in PATH
        return False
    except subprocess.CalledProcessError:
        # Conversion failed
        return False

File: latest_ran_output/qwen3_6_27b/file_004_naive.js
const { exec } = require('child_process');
const fs = require('fs');
const path = require('path');

/**
 * Resizes an image file using the `convert` command (ImageMagick).
 * 
 * @param {string} filePath - The path to the uploaded image file.
 * @param {number} width - The desired width of the resized image.
 * @param {number} height - The desired height of the resized image.
 * @returns {Promise<string>} - The path to the resized image file.
 */
async function resizeImage(filePath, width = 800, height = 600) {
    return new Promise((resolve, reject) => {
        if (!fs.existsSync(filePath)) {
            return reject(new Error('File does not exist'));
        }

        const dir = path.dirname(filePath);
        const baseName = path.basename(filePath, path.extname(filePath));
        const ext = path.extname(filePath);
        const resizedFilePath = path.join(dir, `${baseName}_resized${ext}`);

        // Using ImageMagick's convert command to resize the image
        const command = `convert "${filePath}" -resize ${width}x${height} "${resizedFilePath}"`;

        exec(command, (error, stdout, stderr) => {
            if (error) {
                return reject(new Error(`Error resizing image: ${stderr}`));
            }
            resolve(resizedFilePath);
        });
    });
}

module.exports = { resizeImage };